fastf1/api.py

```diff
@@ -318,16 +322,42 @@ def _write_cache(cls, data, cache_file_path, **kwargs):
             pickle.dump(new_cached, cache_file_obj)
 
     @classmethod
-    def _show_not_enabled_warning(cls):
+    def _enable_default_cache(cls):
         if not cls._CACHE_DIR and not cls._has_been_warned:
-            # warn only once and only if cache is not enabled
-            logging.warning(
-                "\n\nNO CACHE! Api caching has not been enabled! \n\t"
-                "It is highly recommended to enable this feature for much "
-                "faster data loading!\n\t"
-                "Use `fastf1.Cache.enable_cache('path/to/cache/')`\n")
+            cache_dir = None
+            if "FASTF1_CACHE" in os.environ:
+                cache_dir = os.environ.get("FASTF1_CACHE")
+            else:
+                if sys.platform == "linux":
+                    cache_dir = "~/.cache/fastf1"
+                elif sys.platform == "darwin":
+                    cache_dir = "~/Library/Caches/fastf1"
+                elif sys.platform == "win32":
+                    cache_dir = "%LOCALAPPDATA%/Temp/fastf1"
+
+            if cache_dir is not None:
+                # Ensure the default cache folder exists
+                cache_dir = os.path.expandvars(cache_dir)
+                cache_dir = os.path.expanduser(cache_dir)
+                if not os.path.exists(cache_dir):
+                    os.makedirs(cache_dir)
```

Comment:
Given the user can delete the .cache folder it's not guaranteed to exist.

If you'd prefer I would implement a Linux specific check: if ~/.cache exists use it, otherwise just place it in ~/.

Reply:
If `.cache` exists
```
python3
Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fastf1; session = fastf1.get_session(2022, 13, 'Q')
api         WARNING 	

DEFAULT CACHE ENABLED!
	Cache directory: /home/oscar/.cache/fastf1.
	Size: 32.0 KB
>>> 
```

If it doesn't
```
python3
Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import fastf1; session = fastf1.get_session(2022, 13, 'Q')
api         WARNING 	

DEFAULT CACHE ENABLED!
	Cache directory: /home/oscar/.fastf1.
	Size: 321.44 MB
>>> 
```

Reply:
I'm fine with that. It doesn't really make the logic too much more complicated.
Reagarding Windows, while one can in theory delete the temp folder, one really shouldn't. From a bit of quick googling, it sounds like FastF1 breaking might be the least of your concerns then.
The apple documentation phrases it like you can take it as a given that the folder exists. So that should be fine.